From: "Lorenzo Stoakes (ARM)" <ljs@kernel.org>
Date: Mon, 31 Aug 2026 14:42:11 +0100
Subject: [PATCH] mm/vmpressure: remove window size TODO

There has been a steady stream of patches that have been submitted by
newcomers to core mm 'fixing' this TODO, with all but the original having
very likely been generated by LLMs.

It appears that TODOs to LLMs are like red rags to a bull.

In addition, TODOs in code often bitrot and are distracting - those who
understand the code know what could be improved in future.

Therefore remove the TODO.

The work required to actually fix this TODO requires somebody who both has
understanding of the code and significant real-world data to back their
changes.

Such a person doesn't require a TODO prompt to implement this change, so
nothing of value is being lost here.

Link: https://lore.kernel.org/all/20260831130316.448-1-tahasezer.is@gmail.com/
Link: https://lore.kernel.org/linux-mm/20260724054305.516126-1-cui.tao@linux.dev/
Link: https://lore.kernel.org/linux-mm/20260715143646.15828-1-gaikwad.dcg@gmail.com/
Link: https://lore.kernel.org/all/20260227221555.29969-1-mcq@disroot.org/
Link: https://lore.kernel.org/20260831-remove-vmpressure-todo-v1-1-498515e59cdf@kernel.org
Signed-off-by: Lorenzo Stoakes (ARM) <ljs@kernel.org>
Acked-by: Vlastimil Babka (SUSE) <vbabka@kernel.org>
Cc: David Hildenbrand <david@kernel.org>
Cc: Liam R. Howlett <liam@infradead.org>
Cc: Michal Hocko <mhocko@suse.com>
Cc: Mike Rapoport <rppt@kernel.org>
Cc: Suren Baghdasaryan <surenb@google.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---
 mm/vmpressure.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/mm/vmpressure.c b/mm/vmpressure.c
index 9629240d77adc..3de99fef39289 100644
--- a/mm/vmpressure.c
+++ b/mm/vmpressure.c
@@ -30,9 +30,6 @@
  *
  * As the vmscan reclaimer logic works with chunks which are multiple of
  * SWAP_CLUSTER_MAX, it makes sense to use it for the window size as well.
- *
- * TODO: Make the window size depend on machine size, as we do for vmstat
- * thresholds. Currently we set it to 512 pages (2MB for 4KB pages).
  */
 const unsigned long vmpressure_win = SWAP_CLUSTER_MAX * 16;
 
-- 
2.47.3

